sonar-application/src/main/java/org/sonar/application/Cluster.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.JoinConfig;
+import com.hazelcast.config.NetworkConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Manager for the cluster communication between Main Processes
+ */
+public class Cluster implements AutoCloseable {
+  /**
+   * The Hazelcast instance.
+   */
+  @VisibleForTesting
+  public HazelcastInstance hazelcastInstance;
+
+  /**
+   * Instantiates a new Cluster.
+   *
+   * @param clusterProperties The properties of the cluster read from configuration
+   */
+  protected Cluster(ClusterProperties clusterProperties) {
+    if (clusterProperties.isEnabled()) {
+      initializeHazelcast(clusterProperties);
+    }
+  }
+
+  private void initializeHazelcast(ClusterProperties clusterProperties) {
```

Comment:
The validation of parameters is done in ClusterProperties.validate().
I'll add more, however check of all parameters will be hard to implement. For instance, we should check that at least one interface for binding is present, check that members are either IPv4 addresses or resolving DNS name (this is done by the low-level API of Java).
Anyway I'll add more validation in ClusterProperties.validate() method.